lib/facility.rb

```diff
@@ -1,14 +1,79 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles, 
+              :collected_fees 
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  attr_writer :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles, 
+              :collected_fees
+
+  def initialize(facility_details = {})
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def assign_plate_type(vehicle)
+    if vehicle.antique?
+      vehicle.plate_type = :antique
+    elsif vehicle.electric_vehicle?
+      vehicle.plate_type = :ev
+    else
+      vehicle.plate_type = :regular
+    end
+  end
+
+  def register_vehicle(vehicle)
+    if @services.include?("Vehicle Registration")
+        vehicle.registration_date = Date.today
+        assign_plate_type(vehicle)
+        calculate_fees(vehicle)
+        @registered_vehicles << vehicle
+    else
+      false
+    end
+  end
+
+  def calculate_fees(vehicle)
```

Comment:
LOVE the SRP here!! Plate type, fees, and registration are all handled in their own spaces